src/data/finset/slice.lean

```diff
@@ -94,10 +116,20 @@ lemma eq_of_mem_slice (h₁ : A ∈ 𝒜 # r₁) (h₂ : A ∈ 𝒜 # r₂) : r
 lemma ne_of_mem_slice (h₁ : A₁ ∈ 𝒜 # r₁) (h₂ : A₂ ∈ 𝒜 # r₂) : r₁ ≠ r₂ → A₁ ≠ A₂ :=
 mt $ λ h, (sized_slice h₁).symm.trans ((congr_arg card h).trans (sized_slice h₂))
 
-variables [decidable_eq α]
-
-lemma pairwise_disjoint_slice : (set.univ : set ℕ).pairwise_disjoint (slice 𝒜) :=
+lemma pairwise_disjoint_slice [decidable_eq α] : (set.univ : set ℕ).pairwise_disjoint (slice 𝒜) :=
 λ m _ n _ hmn, disjoint_filter.2 $ λ s hs hm hn, hmn $ hm.symm.trans hn
 
+variables [fintype α]
+
+@[simp] lemma bUnion_slice [decidable_eq α] (𝒜 : finset (finset α)) :
+  (range $ fintype.card α + 1).bUnion 𝒜.slice = 𝒜 :=
```

Comment:
(Unrelated to your work but it took me a moment to realise why there was a `+ 1` inside the `range`. Probably this has already been discussed but I wonder if we could also have a `finset.range'` that included the endpoint. I note that the regex `range.*\+ 1` matches 224 results, most looking genuine.)

Reply:
It is called `finset.Iic`!